lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -78,6 +84,9 @@ def exists?
     Puppet.debug "PowerShell Version: #{version}"
     script_content = ps_script_content('test')
     Puppet.debug "\n" + script_content
+    
+    fail self.class.POWERSHELL_UPGRADE_MSG if !PuppetX::Dsc::PowerShellManager.compatible_version_of_powershell?
```

Comment:
Minor nit:  Ruby usually uses `unless` instead of `if not <blah>`  Not sure it improves readability though

```
fail self.class.POWERSHELL_UPGRADE_MSG unless PuppetX::Dsc::PowerShellManager.compatible_version_of_powershell?
```